substrate/frame/contracts/src/wasm/runtime.rs

```diff
@@ -244,6 +254,30 @@ pub enum RuntimeCosts {
 	UnlockDelegateDependency,
 }
 
+macro_rules! cost_write {
+	// cost_write!(name, a, b, c) -> T::WeightInfo::name(a, b, c).saturating_add(T::WeightInfo::rollback_transient_storage())
+	// .saturating_add(T::WeightInfo::set_transient_storage_full().saturating_sub(T::WeightInfo::set_transient_storage_empty())
```

Comment:
Not clear to me why the cost is full - empty, can we explain the reasoning here

Reply:
`full - empty` represents the weight of adding an item to a full BTreeMap. All transient storage host function benchmarks are done with an empty transient storage (BTreeMap), so this represents the worst-case scenario.

Reply:
follow up question, should we refund the cost of rollback_transient_storage in the happy path since you pre-charge it

Reply:
The clear_transient_storage function modifies the storage. The benchmark is currently done for one item in the BTreeMap, but we would like to consider the worst-case scenario, where the modification is done on a fully filled BTreeMap.

I am treating it as a constant cost, similar to DBWrite, which is 100µs.

Reply:
Make sense then, we should add that to the comment to make it clearer

Reply:
Added